src/document.c

```diff
@@ -875,12 +908,36 @@ const char *DocumentField_GetValueCStr(const DocumentField *df, size_t *len) {
     case FLD_VAR_T_CSTR:
       *len = df->strlen;
       return df->strval;
+    case FLD_VAR_T_ARRAY:
+      if (df->arrayLen > 0) {
```

Comment:
I guess this is where we return only the first entry that we need to fix in future PR's, maybe add a comment here that explains it and write a todo?

Reply:
There is a comment below 👇🏼 `// Return the first entry`
And a comment in the header `/* return value as c string (if array - return the first entry)*/`

I can add a TODO for completion of multi value handling (sorting, ...) but there would be many places that will require changes, so I avoided adding many TODOs.
WDYT?